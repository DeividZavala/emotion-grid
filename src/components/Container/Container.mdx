---
name: Container
---

import { Props } from 'docz';
import { Container } from './';

# Container

<Container debug />
