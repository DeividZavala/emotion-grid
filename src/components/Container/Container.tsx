/** @jsx jsx */
import { FC } from 'react';

import styled from '@emotion/styled';
import { jsx, css } from '@emotion/core';

import { config, constants } from '../../config';
import { media } from '../../utils';
import { Types } from '../../types';

import { Row } from '../Row';
import { Col } from '../Col';

const baseStyle = ({ theme }: Types.StyleProps) => css`
  label: container;

  max-width: 100%;

  margin-right: auto;
  margin-left: auto;

  box-sizing: border-box;

  ${constants.BREAKPOINTS.map(
    (breakpoint) => css`
      ${media(breakpoint)} {
        padding-left: ${config(theme).padding[breakpoint] / 2}rem;
        padding-right: ${config(theme).padding[breakpoint] / 2}rem;
      }
    `
  )}
`;

const fluidStyle = ({
  theme,
  fluid,
}: Types.StyleProps & Types.ContainerProps) =>
  !fluid &&
  constants.BREAKPOINTS.map(
    (breakpoint) => css`
      ${media(breakpoint)} {
        ${typeof config(theme).container[breakpoint] === 'number'
          ? `width: 100%; max-width: ${config(theme).container[breakpoint]}rem;`
          : `width: 100%;`}
      }
    `
  );

const debugStyle = ({
  theme,
  debug,
}: Types.StyleProps & Types.ContainerProps) =>
  debug &&
  css`
    ${Row} {
      background: rgba(${config(theme).debug.color}, 0.05);
    }

    ${Col} {
      background: rgba(${config(theme).debug.color}, 0.05);
      border: 1px solid rgb(${config(theme).debug.color});
    }
  `;

const BaseContainer: FC<Types.ContainerProps> = ({ className, children }) => (
  <div className={className}>{children}</div>
);

export const Container = styled(BaseContainer)<Types.ContainerProps>(
  baseStyle,
  fluidStyle,
  debugStyle
);
