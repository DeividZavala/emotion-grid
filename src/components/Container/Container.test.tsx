import React from 'react';
import { render } from '@testing-library/react';

import { Container } from './Container';

describe('Container', () => {
  it('should render with default styles', () => {
    const { container } = render(<Container />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with fluid styles when passed the fluid prop', () => {
    const { container } = render(<Container fluid />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with debug styles when passed the debug prop', () => {
    const { container } = render(<Container debug />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
