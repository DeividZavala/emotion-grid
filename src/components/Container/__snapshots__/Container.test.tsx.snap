// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render with debug styles when passed the debug prop 1`] = `
.grid-4 {
  max-width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:1em) {
  .grid-4 {
    width: 100%;
  }
}

@media (min-width:33.75em) {
  .grid-4 {
    width: 100%;
  }
}

@media (min-width:48em) {
  .grid-4 {
    width: 100%;
    max-width: 45rem;
  }
}

@media (min-width:64em) {
  .grid-4 {
    width: 100%;
    max-width: 60rem;
  }
}

@media (min-width:71.25em) {
  .grid-4 {
    width: 100%;
    max-width: 71.25rem;
  }
}

.grid-4 .grid-3 {
  background: rgba(4,112,245,0.05);
}

.grid-4 .grid-1 {
  background: rgba(4,112,245,0.05);
  border: 1px solid rgb(4,112,245);
}

.grid-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

.grid-0 {
  position: relative;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  width: 100%;
  max-width: 100%;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

<div
  class="grid-4 grid-5"
>
  <div
    class="grid-2 grid-3"
  >
    <div
      class="grid-0 grid-1"
    />
  </div>
</div>
`;

exports[`Container should render with default styles 1`] = `
.grid-0 {
  max-width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:1em) {
  .grid-0 {
    width: 100%;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    width: 100%;
  }
}

@media (min-width:48em) {
  .grid-0 {
    width: 100%;
    max-width: 45rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    width: 100%;
    max-width: 60rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    width: 100%;
    max-width: 71.25rem;
  }
}

<div
  class="grid-0 grid-1"
/>
`;

exports[`Container should render with fluid styles when passed the fluid prop 1`] = `
.grid-0 {
  max-width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media (min-width:1em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:33.75em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:48em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:64em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width:71.25em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

<div
  class="grid-0 grid-1"
/>
`;
